Seda/Seda/Crypto.swift

```diff
@@ -0,0 +1,216 @@
+//  Created by Ryland Sepic on 5/24/20.
+//
+//  This file implements CryptoKit in an easy to use fashion
+//  It can be called as needed to encrypt and decrypt data
+
+import CryptoKit
+import Foundation
+
+// The Crypto object should be created and used to maintain keys and call functions
+
+// NOTE: ~~Currently~~ only one private key is created
+
+/* Steps in order to encrypt your data:
+    1. Generate a private key and store it in the enclave of the device
+ 
+ */
+class Crypto {
+    // This is information on the keys being used
+    // It is set in the code and is needed to create the key and
+    // to decipher the public key when it is read as a string from the database
+    // We want to use an EC type key because it is 256 bits so it can be stored in the enclave
+    let keyType = kSecAttrKeyTypeEC    // Choose cryptography key to use (ex: RSA, EC, etc.)
+    let keySize = 256  // Size of the key in bits
+    let keyTag_raw: Data? = "SedaKey".data(using: .utf8)   // This is the name of the key to retrieve it from the key chain.
+    let algorithm: SecKeyAlgorithm = .eciesEncryptionCofactorVariableIVX963SHA256AESGCM   // The algorithm used to protect the message
+    
+    let keyTag: Data
+    
+    // Keys
+    var priv_key: SecKey? = nil
+    var pub_key: SecKey? = nil
+    
+    init() {
+        guard let tag = self.keyTag_raw else {
+            print("SOMETHING IS VERY WRONG: Can't unwrap a standard string in Crypto initializer")
+            // The keyTag needs to be initialized in the event that this cannot be unwrapped
+            // Something would have to be very wrong for us to end up here
+            let bytesPointer = UnsafeMutableRawPointer.allocate(byteCount: 1, alignment: 1)
+            self.keyTag = Data(bytes: bytesPointer, count: 1)
+            return
+        }
+
+        self.keyTag = tag
+        
+        // Load the private key, if it doesn't exist, create a new one
+        loadPrivKey()
+        if priv_key == nil {
+            createPrivKey() // Create the private key
+        }
+    }
+    
+    
+    // Creates a private key and stores it in the enclave
+    func createPrivKey() {
+        let attributes: [String: Any] = [
+            kSecAttrKeyType         as String:   keyType,
+            kSecAttrKeySizeInBits   as String:   keySize,
+            kSecAttrTokenID         as String:   kSecAttrTokenIDSecureEnclave, // Store the key in the enclave
+            kSecPrivateKeyAttrs     as String: [
+                kSecAttrIsPermanent     as String: true,   // Store the key in the keychain, so we will be able to get the key when the user logs back into the application
+                kSecAttrApplicationTag  as String : keyTag
+                
+            ]
+        ]
+        
+        var err: Unmanaged<CFError>?
+        guard let privateKey = SecKeyCreateRandomKey(attributes as CFDictionary, &err) else {
```

Comment:
I'm getting "Thread 1: EXC_BAD_ACCESS (code=1, address=0x18)" at this line.

Might be related to this: https://stackoverflow.com/questions/12437605/thread-1-exc-bad-access-code-1-address-0x30000008-issue-generated